addons/mail/static/src/bugfix/bugfix.js

```diff
@@ -77,3 +77,77 @@ function useUpdate({ func, priority }) {
 return useUpdate;
 
 });
+
+// Should be moved to its own file in master.
+odoo.define('mail/static/src/component_hooks/use_should_update_based_on_props/use_should_update_based_on_props.js', function (require) {
+'use strict';
+
+const { Component } = owl;
+
+/**
+ * Compares `a` and `b` up to the given `compareDepth`.
+ *
+ * @param {any} a
+ * @param {any} b
+ * @param {Object} compareDepth
+ * @returns {boolean}
+ */
+function isEqual(a, b, compareDepth) {
+    const keys = Object.keys(a);
+    if (Object.keys(b).length !== keys.length) {
+        return false;
+    }
+    for (const key of keys) {
+        // the depth can be given either as a number (for all keys) or as
+        // an object (for each key)
+        let depth;
+        if (typeof compareDepth === 'number') {
+            depth = compareDepth;
```

Comment:
doesn't happen based on method docstring, and also in practice never used by providing number

Reply:
happens on recursive calls, at least the "final" depth will be always called with 0

Reply:
Ok, so `compareDepth` is either `Object` or `number`. Just need to update docstring.

Reply:
indeed